components/api/src/main/java/com/hotels/styx/api/StyxCoreObservable.java

```diff
@@ -92,7 +94,7 @@ public StyxCoreObservable(CompletionStage<T> future) {
     }
 
     private static <T> CompletableFuture<T> fromSingleObservable(Observable<T> observable) {
-        final CompletableFuture<T> future = new CompletableFuture<>();
```

Comment:
Why do you need to remove the final modifier?

Reply:
Just keeping it in line with our standard coding practices. Excessive `final`s clutter the codebase, so we avoid them on local variables and method parameters (due to their limited scope). 
We only use them on fields because it could be very hard to tract down an unexpected mutation in a large class.
For locals and parameters, we instead aim to keep our methods small and simple enough that any mutations are easily visible (in addition to the other benefits of readable methods).